### PR TITLE
refactor(tests): use fixture and parammeterized pytest

### PR DESCRIPTION
**Pull request by bwanglzu**
This is a Proof of concept PR (not intended to be merged) on how should we refactor duplicated tests with:

1. pytest fixtures.
2. pytest parameterized mark.
3. Replace `assertEqual(a, b)` syntax with `assert a==b` which is more pythonic.

Besides, class `unittest` style of class based test could be replace by function based test.

### CODE REVIEW

File: tests/unit/drivers/test_chunk2doc_rank_drivers.py [hunk 1]
```diff
@@ -41,62 +43,46 @@ def create_document_to_score():
...
+    scope="module",
+    params=[
+        MockLengthRanker(),
+        MaxRanker(),
+        MinRanker()
+])
+def ranker(request):
+    return request.param
+
+@pytest.fixture(scope="module")
```
Comment (JoanFM) [inline tests/unit/drivers/test_chunk2doc_rank_drivers.py:56]: what other options of scope are there?

**Comment by hanxiao** [comment]
@bwanglzu can you sign CLA for this PR, i already added to the whitelist

**Comment by bwanglzu** [comment]
@hanxiao thanks! I'll do it this morning

File: tests/unit/drivers/test_chunk2doc_rank_drivers.py (hunk 1 above)
Comment (bwanglzu) [reply tests/unit/drivers/test_chunk2doc_rank_drivers.py:56]: @JoanFM Possible values for scope are: function, class, module, package or session.

**Comment by bwanglzu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by JoanFM** [comment]
LGTM! I will need to get used to this but I guess it is better to keep best practices and standards.

**Comment by hanxiao** [comment]
recheckcla

**Comment by hanxiao** [comment]
> refactor(tests): Remove unused lines, and parameterized argument

this commit msg is too long

Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)

**Comment by nan-wang** [comment]
@bwanglzu Would you please rephrase the commit messages?

**Comment by bwanglzu** [comment]
I'll close this one and send a new one a bit later

**Pull request closed without merging**